### PR TITLE
[Feature: Notification] Course material notifications

### LINKED ISSUE #2490: Add notification badge to course materials
(# of new files since you last viewed)

See also #2489

### PR DESCRIPTION
**Pull request by mukul-kmr-jha**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
No notification is given for course material upload events.

### What is the new behavior?
Closes #2490 
Whenever there are new course material files added for a user (section-specific or for everyone), there is this badge shown to the users on the course material nav-link showing the number of new files for them to see.

### Other information?
course_material_badge

### CODE REVIEW

File: migration/migrator/migrations/course/20200408021149_course_material_notification_info.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute('CREATE TABLE course_material_info (user_id character varying(255) NOT NULL,course_file_path TEXT NOT NULL,release_date timestamp with time zone NOT NULL,seen BOOLEAN DEFAULT FALSE NOT NULL)')
```
[the remark this answers is not in the document]
Comment (mukul-kmr-jha) [reply migration/migrator/migrations/course/20200408021149_course_material_notification_info.py:17]: So in this PR, running a down migration will not break things. But infact things get into broken state if proper down script are not written.

File: site/public/js/server.js [hunk 2]
```diff
@@ -1511,7 +1511,7 @@ function changeNewDateTime(filename, newdatatime,handleData) {
                 alert("ERROR: Invalid date.");
                 return false;
             }
-
+            debugger;
```
[the remark this answers is not in the document]
Comment (mukul-kmr-jha) [reply site/public/js/server.js:1514]: Let me remove this in my next commit with the other pending changes. Thanks!

File: migration/migrator/migrations/course/20200408021149_course_material_notification_info.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MasterOdin) [reply migration/migrator/migrations/course/20200408021149_course_material_notification_info.py:17]: In an ideal world, all code we write is perfect and works flawlessly and never breaks things. The real world is unfortunately a bit messier, in that sometimes PRs get merged, and find that they've got some showstopper bug causing us to revert it, and so the down method will get run, and then once things are fixed, we'll re-run the up migration.

For "standalone" patches such as this one where you're adding a new table only, it's fine to just wrap your execute functions with a `if not database.has_table('course_material_info')` or use a `IF NOT EXISTS` modifier on the queries as that way we will not lose any data inserted into these tables.

If you were on the other-hand altering existing columns in a database, the down function would be expected to reverse those transformations.

[the remark this answers is not in the document]
Comment (MasterOdin) [reply migration/migrator/migrations/course/20200408021149_course_material_notification_info.py:17]: See https://submitty.org/developer/migrations#writing-new-migrations (especially notes at the bottom) for discussion on this.

[the remark this answers is not in the document]
Comment (mukul-kmr-jha) [reply migration/migrator/migrations/course/20200408021149_course_material_notification_info.py:17]: I'm on it.. Will keep this in mind.
Thanks!

**Comment by bmcutler** [comment]
Thanks for the PR @mukul-kmr-jha !

But we're closing this because course materials needs a larger refactor & redesign and this particular feature is lower priority.

**Pull request closed without merging**